src/Nerdbank.MessagePack/MessagePackPrimitiveSpanUtility.cs

```diff
@@ -0,0 +1,1831 @@
+				results.StoreUnsafe(ref Unsafe.As<byte, sbyte>(ref output), offset);
+			}
+
+			{
+				offset = length - unchecked((nuint)Vector<sbyte>.Count);
+				Vector<sbyte> results = Vector.Equals(Vector.LoadUnsafe(ref Unsafe.As<bool, sbyte>(ref input), offset), Vector<sbyte>.Zero) + new Vector<sbyte>(unchecked((sbyte)MessagePackCode.True));
+				results.StoreUnsafe(ref Unsafe.As<byte, sbyte>(ref output), offset);
+			}
+		}
+		else
+		{
+			for (; offset < length; offset++)
+			{
+				Unsafe.Add(ref output, offset) = Unsafe.Add(ref input, offset) ? MessagePackCode.True : MessagePackCode.False;
+			}
+		}
+	}
+
+	/// <summary>
+	/// Write <see cref="sbyte"/> values.
+	/// </summary>
+	/// <param name="output">The reference to the <see cref="byte"/> buffer which receives the serialized values.</param>
+	/// <param name="input">The reference to the source <see cref="sbyte"/> buffer.</param>
+	/// <param name="inputLength">The length of the input/output buffers.</param>
+	/// <returns>Written <see cref="byte"/> count.</returns>
+	public static nuint Write(ref byte output, ref sbyte input, nuint inputLength)
+	{
+		nuint inputOffset = 0, outputOffset = 0;
+		if (Vector.IsHardwareAccelerated && inputLength >= unchecked((nuint)Vector<sbyte>.Count))
+		{
+			do
+			{
+				var loaded = Vector.LoadUnsafe(ref input, inputOffset);
+				var additionalBytes = Vector.GreaterThan(new Vector<sbyte>(unchecked((sbyte)MessagePackCode.MinNegativeFixInt)), loaded);
+				if (additionalBytes == Vector<sbyte>.Zero)
+				{
+					loaded.As<sbyte, byte>().StoreUnsafe(ref output, outputOffset);
+					inputOffset += unchecked((nuint)Vector<sbyte>.Count);
+					outputOffset += unchecked((nuint)Vector<sbyte>.Count);
+					continue;
+				}
+
+				for (int i = 0; i < Vector<sbyte>.Count; i++)
+				{
+					if (additionalBytes.GetElement(i) != 0)
+					{
+						Unsafe.Add(ref output, outputOffset++) = MessagePackCode.Int8;
+					}
+
+					Unsafe.Add(ref output, outputOffset++) = unchecked((byte)Unsafe.Add(ref input, inputOffset++));
+				}
+			}
+			while (inputOffset + unchecked((nuint)Vector<sbyte>.Count) <= inputLength);
+		}
+
+		for (; inputOffset < inputLength; inputOffset++)
+		{
+			sbyte temp = Unsafe.Add(ref input, inputOffset);
+			if (temp < MessagePackRange.MinFixNegativeInt)
+			{
+				Unsafe.Add(ref output, outputOffset++) = MessagePackCode.Int8;
+			}
+
+			Unsafe.Add(ref output, outputOffset++) = unchecked((byte)temp);
+		}
+
+		return outputOffset;
+	}
+
+	/// <summary>
+	/// Write <see cref="short"/> values.
+	/// </summary>
+	/// <param name="output">The reference to the <see cref="byte"/> buffer which receives the serialized values.</param>
+	/// <param name="input">The reference to the source <see cref="short"/> buffer.</param>
+	/// <param name="inputLength">The length of the input/output buffers.</param>
```

Comment:
The buffers are made up of elements with different individual lengths. We should be careful to document whether this is the *count* of elements, or the byte length of the buffers.

Reply:
Thank you for rewriting and fixing!